---
# Skill Check
---
Please load the dataset "Compustat_fy2019.csv" and use the functions we learned in class to answer the following questions. You can run the code in below empty cells and fill the answers in the text box on Canvas. Please save and upload your notebook (the .ipynb file) to the Canvas assignment by attaching it in the text box.

**1. What is the shape of the data frame? (Please indicate how many observations and variables there are.)**

**2. What is the net income for 29th observation of "Compustat_fy2019.csv" (AMERICAN EXPRESS CO)?** 

**3. Among companies listed in NASDAQ, which company has the largest total equities according to the dataset?**  
Note: companies listed in NASDAQ has an exchange code of 14.  
(This data set includes companies listed in three exchanges. They are New York Stock Exchange (11), American Stock Exchange (12), NASDAQ (14).  

**4. What's the index that Python has assigned to the observation for the company that you found in question 3?** 


#### Reference: Variable Definitions
- The column names represent the measures in parentheses:
    - tic (ticker symbol)
    - conm (company name)
    - datadate (fiscal year end)
    - fyear (fiscal year)
    - at (total assets)
    - lt (total liabilities)
    - teq (total equities)
    - revt (revenue)
    - ni (net income)
    - exchg (numeric exchange code; 11 = New York Stock Exchange; 12 = American Stock Exchange; 14 = NASDAQ)

In [1]:
import pandas as pd

In [3]:
df = pd.read_csv("Compustat_fy2019.csv", parse_dates = ['datadate'])

/var/folders/x8/fqk_hz6j4x1f9s70rdmxj0d00000gn/T/ipykernel_73625/2995662502.py:1: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df = pd.read_csv("Compustat_fy2019.csv", parse_dates = ['datadate'])


## 1. What is the shape of the data frame? (Please indicate how many observations and variables there are.)

In [7]:
df.shape

(4727, 10)

## 2. What is the net income for 29th observation of "Compustat_fy2019.csv" (AMERICAN EXPRESS CO)?

In [19]:
df.iloc[28]

tic                         AXP
conm        AMERICAN EXPRESS CO
datadate    2019-12-31 00:00:00
fyear                      2019
at                     198321.0
lt                     175250.0
teq                     23071.0
revt                    47020.0
ni                       6759.0
exchg                        11
Name: 28, dtype: object

The net income for AMEX is 6759

## 3. Among companies listed in NASDAQ, which company has the largest total equities according to the dataset?

In [18]:
filtered_df = df[df['exchg'] == 14]
sorted_df = filtered_df.sort_values(by='teq', ascending=False)
sorted_df

tic                     conm   datadate  fyear          at  \
3819  GOOGL             ALPHABET INC 2019-12-31   2019  275909.000   
745    MSFT           MICROSOFT CORP 2019-06-30   2019  286556.000   
4025     FB             FACEBOOK INC 2019-12-31   2019  133376.000   
45     AAPL                APPLE INC 2019-09-30   2019  338516.000   
146   CMCSA             COMCAST CORP 2019-12-31   2019  263414.000   
...     ...                      ...        ...    ...         ...   
3327   VRSN             VERISIGN INC 2019-12-31   2019    1854.009   
669    SGMS    SCIENTIFIC GAMES CORP 2019-12-31   2019    7809.000   
2905    PPD                  PPD INC 2019-12-31   2019    5556.246   
3495   SBAC  SBA COMMUNICATIONS CORP 2019-12-31   2019    9759.941   
1758   SBUX           STARBUCKS CORP 2019-09-30   2019   19219.600   

              lt         teq        revt         ni  exchg  
3819   74467.000  201442.000  161857.000  34343.000     14  
745   184226.000  102330.000  125843.000  39240.000     14  
4025   32322.000  101054.000   70697.000  18485.000     14  
45    248028.000   90488.000  260174.000  55256.000     14  
146   178168.000   83874.000  108942.000  13057.000     14  
...          ...         ...         ...        ...    ...  
3327    3344.109   -1490.100    1231.661    612.299     14  
669     9917.000   -2108.000    3400.000   -130.000     14  
2905    8224.358   -2698.148    4031.017     47.821     14  
3495   13410.896   -3667.007    2014.645    146.991     14  
1758   25450.600   -6231.000   26508.600   3599.200     14  

[2673 rows x 10 columns]

## 4. What's the index that Python has assigned to the observation for the company that you found in question 3?

The answer is google which is at index 3819